dask/dataframe/partitionquantiles.py

```diff
@@ -385,7 +386,8 @@ def process_val_weights(vals_and_weights, npartitions, dtype_info):
     return rv
 
 
-def percentiles_summary(df, num_old, num_new, upsample, state):
+@methods.percentiles_summary.register(pd.Series)
+def percentiles_summary_pandas(df, num_old, num_new, upsample, state):
```

Comment:
Is it possible to avoid dispatching this one? 

If we added a `tolist()` method to cudf, and allowed for a very small amount of host-device transfer for some random values, would we need to split this off?  Dispatch functions open up some code complexity.  I would prefer to use them sparingly if we can avoid them.  Obviously in some cases we can't avoid them, but this seems like a case where I'm not sure.  Thoughts?

Reply:
Thanks for looking at this @mrocklin !

I added the `percentiles_summary` dispatch because the `data = df.values` line becomes problematic once the `values` property becomes a cupy array in cudf.

The original problem was that this leads to a "hang" during multi-threaded execution.  Therefore, I was planning to (temporarily) use `data = df.values_host` in the cudf definition. Assuming that the "hang" problem is fixed by numba#4581, the use of a cupy-typed `values` property will still cause a problem in `percentiles_summary`, because the required call to `np.percentile`  does **not** seem to handle cupy input.

Reply:
If the number of new dispatch functions is a concern, it would be easiest for me to combine `group_split` and `group_split_2` (rather than removing `percentiles_summary`) -- Any thoughts on that?